### DISCUSSION TITLE
Persist error on refetch

### DISCUSSION THREAD

**Discussion by Sagie501** [TanStack/query#5246] [Ideas]
Hi!

I think it would be very useful to add `persistError` property to the `useQuery` options.
Today if I had error returned from my query and then the query refetched (doesn't matter why maybe explicit refetch, maybe refetchInterval or any other reason) the `error` is set to `null` and `isError` to false.
Adding `persistError` property and set it to true will keep the `error` state and `isError` true.

I would be happy if you share what you think about it!
Thanks!

**Comment by TkDodo** [comment]
`Error` is not serializable to json so per default, this wouldn't work. You'd need to provide s custom serializer like superjson

**Reply by Sagie501** [reply]
@TkDodo Although it's a bit old do you have an idea what to do to overcome this issue?
I have the same issue as presented here:
https://www.reddit.com/r/react/comments/1hda1rq/persisting_error_state_in_reactquery_until/

**Reply by TkDodo** [reply]
Errorboundary
